flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/functions/SqlDateTimeUtils.java

```diff
@@ -1297,4 +1297,10 @@ private static int parseFraction(String v, int multiplier) {
 		return r;
 	}
 
+	// TODO: remove if CALCITE-3199 fixed
+	//  https://issues.apache.org/jira/browse/CALCITE-3199
+	public static long unixDateCeil(TimeUnitRange range, long date) {
+		return julianDateFloor(range, (int) date + EPOCH_JULIAN, false);
```

Comment:
Is this correct? It still calls `SqlDateTimeUtils#julianDateFloor` not Calcite's `DateTimeutils`.

Reply:
Yes, it is, the Julian flag is what CALCITE-3199 fixed.

Reply:
the implementation of `Is this correct? It still calls SqlDateTimeUtils#julianDateFloor` is different from `DateTimeutils`, but i didn't find any bad case.